### ISSUE TITLE
Many broken themes

### ISSUE THREAD

**Issue by someone-somenet-org** [getpelican/pelican-themes#677]
I just tried to take a look at all the themes ... 
Of all **128** themes here, only these 23 seem to work with pelican `4.2.0`. :/

Could you consider moving old/unmaintained themes to some subdir?

+ aboutwilson
+ attila
+ bootstrap2
+ bootstrap2-dark
+ franticworld
+ martin-pelican
+ materialistic
+ mnmlist
+ monospace
+ new-bootstrap2
+ nice-blog
+ nikhil-theme
+ nmnlist
+ octopress
+ plumage
+ relapse
+ subtle
+ svbhack
+ taman
+ tuxlite_zf
+ voidy-bootstrap
+ w3-personal-blog
+ zurb-F5-basic

**Comment by kakila** [comment]
This would be really needed, I spent a lot of time testing before finding this issue

**Comment by KeithCu** [comment]
It is amazing this issue doesn't get fixed.

**Comment by dakshsriv** [comment]
Apparently, a 24th theme has broken down. twenty-pelican-html5up has worked horribly: [#10 on twenty-pelican-html5up](https://github.com/frankV/twenty-pelican-html5up/issues/10).
Also, it's not pelican's fault that some themes are sub-maintained. It's the theme makers' fault.

**Comment by someone-somenet-org** [comment]
> Also, it's not pelican's fault that some themes are sub-maintained. It's the theme makers' fault.

true, but it looks bad for pelican to host seemingly broken templates giving pelican as a whole a little dead feeling.

It would - for example - be great to get the templates into versioned directories. That way, if you just care about previewing/testing the template, you can downgrade pelican. And you would not have a large list of dead templates, just a versioned list.

**Comment by maphew** [comment]
The picture is a lot better at end of 2022 but there are still far too many broken themes at the top end.  Current stats in my testing are 73 working and 56 not; listed [here](https://github.com/maphew/pelican-themes-preview/blob/main/Themese-not-working.md) (note: short term link).  I'm happy to put some effort at re-arranging, just let me know what I could do.

**Comment by leekimber** [comment]
Following the link from [#645](https://github.com/getpelican/pelican-themes/issues/645)... Thanks maphew for surveying the themes. It's an interesting read. It formalises what I encountered last year when I started looking for a  small file-size, dark/light mode-supporting theme that was as attractive as the [Alchemy (bootstrap) theme](https://github.com/nairobilug/pelican-alchemy/blob/master/README.md).

I concluded that, as a non-web developer pelican enthusiast, there was no suitable theme and that I would have to tackle the compound problem (that is: the underlying complex of problems) of creating a theme myself:

The problems I encountered are:

- Web UI design has moved fast into semantic HTML, multimedia content (HTML5 + media controls) - which are challenging for markdown in its original form as a writer's tool.
- Web UI design talent has adopted css toolchains and js toolchains - which increase the learning-load for many pelican python users like me.
- Web UI design has gone full-on responsive (smartphone as primary client) which means much time spent sifting through existing pelican themes to find decent responsive techniques to copy.
- Web UI design has morphed around the expectations of smartphone users (images and icons over text for link identifiers). This reflects changes in style and navigation expectations that are not reflected in many themes.
- Web UI design has gone Progressive Web App (PWA), which is easily achieved in pelican but not widely documented and add extra thought at design time and a management considerations to your site workflow after first build.
- Designing beautiful reading experiences seems to have lost out to designing interactive web apps generally (read Jeremy Keith's [The State of the Web](https://adactio.com/articles/18580) if you are not familiar with the problems this causes. [This is also good, though long](https://adactio.com/articles/17733).)
- Pelican grew so popular so fast that Jmeyer et al couldn't possibly document answers to all the use cases and workflows that users were asking about.
- Web UI design changed so fast that jmeyer et al couldn't possibly answer or document answers to all the *theme* questions that arise.
- The separation of themes into two collections may not have helped solve all of the above complex of problems.

This one is hard to capture but it seems to me an invisible accident occurred. I think jmeyer et al created us a great python back-end and offered us starter themes and a chance to add more themes. Which people who were not web designers by profession or inclination did. But while they were doing it, the web world moved on a long way in multiple directions - some of them even damaging to web users - leaving pelican themes behind.

There are probably other issues too but even I'm tired of reading me.

Recognising that what I want in a pelican theme may be different to what others want, here's a list of what I want:

- Required: As beautiful as Alchemy. Or, at least, easy on the eye. 
- Required: Device-determined light mode or dark mode plus easy user over-ride. Like the great user experience jenil demonstrates with [Chota](https://jenil.github.io/chota/). See that little 'star' at the bottom right?
- Required: Respect users with old browsers. As in: fail with legibility not with apology.
- Required: Minimum web UI skills required to tweak it. I'm not a designer but I can tweak things.
- Absolutely required: No web UI tools or toolchain required to tweak it. Just .css and .js files laid out on tinkerable individual lines.
- Required: Tiny .css file.
- Optional: No, or small, .js file.
- Optional: PWA support.
- Optional but please...: Documentation for it and its relationship with pelicanconf.py variables.
- Optional: Demo pages that show what it can do, not demo pages that show you abandoned pelican for next.js or gatsby's themes.

I've been trying to solve these problems by:
- Experimenting with lightweight css themes to find one that a non-UI designer can tweak into the above requirements within one human lifespan.
- Continuing the [PWA experiment I began with Alchemy](https://github.com/leekimber/pelican-alchemy-pwa). Don't use that code. I need to clean my latest version and push it, plus finish my documentation for it.
- Writing a markdown pre-processor that lets me research and write in markdown as a writer in a multimedia and HTML5 world. Which means it:
    - recognises multimedia refs in markdown and wraps them in HTML5 tags
    - recognises the writer's intention to caption and acts accordingly. Ie it wraps the caption and its associated media ref in <figure>+<figcaption> tags.
    - applies [<figure>+<figcaption> tags to blockquotes](https://stackoverflow.com/questions/48638937/a-blockquote-in-a-figure-when-to-use-footer-when-to-use-figcaption).

I've tried a lot of CSS themes. I quite like:
- [Chota](https://jenil.github.io/chota/)
- [Inviscss](https://cmroanirgo.github.io/inviscss/)
- [Lotus](http://goatslacker.github.io/lotus.css/)

I would use one of the existing Bulma-based pelican themes if they didn't make images and videos even smaller on mobile than they already are and met my dark/light mode requirement.

I've tried many more than the above. I'm currently focused on Chota because it looks as though I'll eventually get it close to [Alchemy's layout](https://github.com/nairobilug/pelican-alchemy/blob/master/README.md). But with tiny Chota filesizes and jenil's understanding of smartphone users' mindsets.

But what ever I eventually come up with, it won't solve the theme problems maphew identified. 

I wonder if pelican themes management needs an approach that can create and maintain a base set of themes that are: - modern in look
- modern in functionality
-  and respect device trends (smartphones - even though their interface is appalling). 
 
Perhaps it might work to take the best in navigation and functionality trends in HTML dev and multimedia support and require them to be in a core set of themes but not let the core set become dependent on library fashions (bootstrap, jquery, css class conflics, or whatever).

I'm not saying: don't have more complex themes. I am saying: perhaps we need to work towards creating and maintaining a base of modern minimal themes regardless of what css or js or toolchain is the fashion du jour.  And providing support for them. While encouraging users to add to a separate set of more complex or difficult themes for which pelican provides no support.   

This is where I think pelican lost out. I don't know where jmeyer's skill-boundaries are but I do know that work takes time and his work is in providing a fantastic python backend. Meanwhile, the web UI world has added huge workloads in theme design - and even markdown rendering requirements - that no-one has been able to keep pace with.

So I think it might be that we have to figure out a way for us users to find a way to better support a few themes that make pelican look good on modern devices with modern standards. And leave users to maintain their own themes in a separate area. 

I don't think this work can be done entirely by non-UI folks like me but I'm happy to add in my code to that effort or some variation of that effort.

Thoughts?

**Comment by maphew** [comment]
> Thoughts?

Oh yeah, way too many to write out. Lot of affirmative head nodding with regard to the goal of a minimal css starting point yet also friendly for middling html-css knowledgeable folks to build on. Thanks for sharing your extensive thoughts. More later as I go through and digest them.

Also see https://github.com/mosra/m.css, which is a highly refined pelican theme with many of these ideals at heart. It's not referenced in the _pelican-themes_ repository.
